components/DateSelect.js

```diff
@@ -0,0 +1,134 @@
+                    disabled={disabled}
+                    onPress={() => setIsModalOpen(true)}
+                >
+                    {!stagedDate && !!placeholder && <PlaceholderText>{placeholder}</PlaceholderText>}
+                    <PseudoInputText disabled={disabled}>{stagedDate && Format(stagedDate, 'P')}</PseudoInputText>
+                </PseudoInput>
+                {!!caption && caption()}
+            </View>
+            <Modal
+                visible={isModalOpen}
+                onBackdropPress={() => setIsModalOpen(false)}
+            >
+                <DateContainer>
+                    <DateLayout>
+                        <View>
+                            <Calendar
+                                date={stagedDate}
+                                onSelect={(nextDate) => setStagedDate(nextDate)}
+                                {...props}
+                            />
+                        </View>
+                    </DateLayout>
+                    <Divider />
+                    <ButtonWrapper>
+                        <Button
+                            appearance='ghost'
+                            status='basic'
+                            onPress={handlePressCancel}
+                            style={{ marginRight: Theme.spacing(1) }}
+                        >
+                            Cancel
+                        </Button>
+                        <Button
+                            onPress={handlePressOk}
+                        >
+                            Ok
+                        </Button>
+                    </ButtonWrapper>
+                </DateContainer>
+            </Modal>
+        </View>
+    );
+};
+
+internals.DateLayout = Styled.View`
+    flex-direction: row;
+`;
+
+internals.DateContainer = Styled.View`
+    background-color: ${({ theme }) => theme.palette.basic[100]};
+    border-radius: ${({ theme }) => theme.shape.borderRadius}px;
+    border: 2px solid ${({ theme }) => theme.palette.basic[400]};
+    overflow: hidden;
+`;
+
+internals.ButtonWrapper = Styled.View`
+    flex-direction: row;
+    padding: ${({ theme }) => theme.spacing(1,2)};
+    justify-content: flex-end;
+`;
+
+internals.PseudoInput = Styled.Pressable`
+    flex-direction: row;
+    align-items: center;
+    padding: ${({ theme }) => theme.spacing(1)}px;
+    background-color: #eee;
+    border-radius: 4px;
+    border: 1px solid ${({ theme }) => theme.palette.basic[400]};
+    min-height: 40px;
+`;
+
+internals.PseudoInputText = Styled(Text)`
+    color: ${({ disabled, theme }) => (disabled ? theme.palette.basic[600] : 'black')};
+`;
+
+internals.PlaceholderText = Styled(Text)`
+    color: ${({ theme }) => theme.palette.basic[600]};
+`;
+
+module.exports.propTypes = {
+    value: T.instanceOf(Date),
+    onChange: T.func.isRequired,
+    label: T.string,
+    caption: T.func,
+    disabled: T.bool,
+    placeholder: T.string
+};
+
+module.exports.defaultProps = {
+    label: 'Date'
+};
```

Comment:
oh my. I'm so sorry to have you add this. I didn't realize there's this much code for `DateSelect` and thanks for adding it 🙏